_drafts/react-example-todomvc.md

```diff
@@ -0,0 +1,119 @@
+
+#### Mixing the two
+
+The following figure shows what happens when the user interacts with the application (add a new task, click on a checkbox to select a specific task...):
+
+![MVC with React](/img/posts/2015/react-example-todomvc-steps.png)
+
+1. the view sends the action to the controller ;
+2. the controller gets the current model from the reactive signal and build a new model accordingly to the action ;
+3. the controller sets this new model as the new reactive signal value ;
+4. the view detects that a new model is available (the view reacts to the new signal value) and updates itself with the new model data.
+
+Let's now see how this is implemented. Please note that i will only detail the reactive features.
+
+### Initialization
+
+The [main function](https://github.com/slegrand45/examples_ocsigen/blob/d6766d404a449d0b1d36ad3cd916b0c444390a19/jsoo/todomvc-react/todomvc.ml#L393) creates the reactive signal with an initial model (possibly empty). The `m` value is of type `Model.t`:
+
+{% highlight ocaml %}
+let rp = React.S.create m in
+{% endhighlight %}
+
+React.S.create returns a tuple. The first value is a primitive signal. The second value is a function which will be used later by the controller [to set a new model as the new signal value](https://github.com/slegrand45/examples_ocsigen/blob/d6766d404a449d0b1d36ad3cd916b0c444390a19/jsoo/todomvc-react/todomvc.ml#L170).
+
+### Reactive attribute
+
+This first example explains how the CSS style of a HTML node becomes reactive ([source code](https://github.com/slegrand45/examples_ocsigen/blob/d6766d404a449d0b1d36ad3cd916b0c444390a19/jsoo/todomvc-react/todomvc.ml#L267-L299)). In the Todo application, the tasks list is displayed in a `<section>` HTML tag. The CSS style of this HTML node must contain `visibility: hidden;` if the tasks list is empty. But must contain `visibility: visible;` if the number of tasks is greater than zero. So the style attribute of this `<section>` node must change regarding the model content:
+
+{% highlight ocaml %}
+R.Html5.a_style (React.S.map css_visibility r)
+{% endhighlight %}
+
+The first thing to note is that we use the Tyxml_js.R.Html5 module instead of the Tyxml_js.Html5 one. Tyxml_js.R.Html5 is simply the Reactive counterpart of Tyxml_js.Html5. As it's a reactive attribute, the a_style function waits a reactive signal as its argument. Here we use React.S.map which have the signature `('a -> 'b) -> 'a React.signal -> 'b React.signal`. This map function takes as its first argument a function named css_visibility:
```

Comment:
You missed various quote quotation `` ` `` in this paragraph. I think you can say that you use `Tyxml_js` once and then drop it for the rest of the post.